modules/gdscript/gdscript_parser.cpp

```diff
@@ -4035,6 +4039,44 @@ bool GDScriptParser::warning_annotations(const AnnotationNode *p_annotation, Nod
 #endif // DEBUG_ENABLED
 }
 
+bool GDScriptParser::tool_button_annotation(const AnnotationNode *p_annotation, Node *p_node) {
+#ifndef TOOLS_ENABLED // !TOOLS_ENABLED
+	// Only available in editor.
+	return true;
+#else // TOOLS_ENABLED
+	AnnotationNode *annotation = const_cast<AnnotationNode *>(p_annotation);
+	FunctionNode *func = static_cast<FunctionNode *>(p_node);
+	if (annotation->arguments.size() < 1) {
+		push_error("Tool buttons must specify a name.", p_annotation);
+		return false;
+	}
+
+	if (!this->is_tool()) {
+		push_error("Tool buttons can only be used in tool scripts.", p_annotation);
+		return false;
+	}
+
+	annotation->export_info.name = annotation->resolved_arguments[0];
+	annotation->export_info.type = Variant::Type::CALLABLE;
+	annotation->export_info.usage = PROPERTY_USAGE_EDITOR | PROPERTY_USAGE_INTERNAL;
+	if (func->parameters.is_empty()) {
```

Comment:
This should check if the argument count is 1, because the call fails silently when there is more arguments.
(and then the error should probably say `Tool button method must take exactly 1 argument.`)